### PR TITLE
docs/all: document callback parameters

### PR DESCRIPTION
**Pull request by m-cat**

### CODE REVIEW

File: safe_app/src/ffi/mutable_data/permissions.rs [hunk 1]
```diff
@@ -72,7 +72,9 @@ impl Into<Action> for MDataAction {
 pub unsafe extern "C" fn mdata_permission_set_new(
     app: *const App,
     user_data: *mut c_void,
-    o_cb: extern "C" fn(*mut c_void, FfiResult, MDataPermissionSetHandle),
+    o_cb: extern "C" fn(user_data: *mut c_void,
+                        result: FfiResult,
+                        MDataPermissionSetHandle),
```
Comment (nbaksalyar) [inline safe_app/src/ffi/mutable_data/permissions.rs]: Missed `perm_set_h: MDataPermissionSetHandle`

File: safe_app/src/ffi/mutable_data/permissions.rs [hunk 2]
```diff
@@ -144,7 +146,7 @@ pub unsafe extern "C" fn mdata_permission_set_is_allowed(
     set_h: MDataPermissionSetHandle,
     action: MDataAction,
     user_data: *mut c_void,
-    o_cb: extern "C" fn(*mut c_void, FfiResult, PermissionValue),
+    o_cb: extern "C" fn(user_data: *mut c_void, result: FfiResult, PermissionValue),
```
Comment (nbaksalyar) [inline safe_app/src/ffi/mutable_data/permissions.rs]: Missed `value: PermissionValue`

File: safe_app/src/ffi/nfs.rs [hunk 3]
```diff
@@ -227,14 +244,19 @@ pub unsafe extern "C" fn file_size(
...
 pub unsafe extern "C" fn file_read(
     app: *const App,
     file_h: FileContextHandle,
     position: u64,
     len: u64,
     user_data: *mut c_void,
-    o_cb: extern "C" fn(*mut c_void, FfiResult, *const u8, usize),
+    o_cb: extern "C" fn(user_data: *mut c_void,
+                        result: FfiResult,
+                        file_data_ptr: *const u8,
```
Comment (nbaksalyar) [inline safe_app/src/ffi/nfs.rs]: Probably just `data_ptr` and `data_len`?

File: safe_authenticator/src/ffi/apps.rs [hunk 4]
```diff
@@ -111,12 +113,16 @@ pub unsafe extern "C" fn auth_rm_revoked_app(
     });
 }
 
-/// Get a list of apps revoked from authenticator
-#[no_mangle]
+/// Get a list of apps revoked from authenticator.
+///
+/// Callback parameters: user data, error code, app exchange info
```
Comment (nbaksalyar) [inline safe_authenticator/src/ffi/apps.rs]: `..., app exchange info vector, and vector size`

File: safe_authenticator/src/ffi/apps.rs [hunk 5]
```diff
@@ -111,12 +113,16 @@ pub unsafe extern "C" fn auth_rm_revoked_app(
     });
 }
 
-/// Get a list of apps revoked from authenticator
-#[no_mangle]
```
Comment (nbaksalyar) [inline safe_authenticator/src/ffi/apps.rs]: `#[no_mangle]` got accidentally removed

File: safe_authenticator/src/ffi/apps.rs [hunk 6]
```diff
@@ -251,13 +262,18 @@ pub unsafe extern "C" fn auth_registered_apps(
 
 /// Return a list of apps having access to an arbitrary MD object.
 /// `md_name` and `md_type_tag` together correspond to a single MD.
+///
+/// Callback parameters: user data, error code, app access
```
Comment (nbaksalyar) [inline safe_authenticator/src/ffi/apps.rs]: `app access, app access len`

File: safe_authenticator/src/ffi/mod.rs [hunk 7]
```diff
@@ -79,14 +83,18 @@ pub unsafe extern "C" fn create_acc(
...
 /// to the first parameter of the network events observer callback (`o_network_obs_cb`).
+///
+/// Callback parameters: user data, error code, authenticator
 #[no_mangle]
 pub unsafe extern "C" fn login(
     account_locator: *const c_char,
     account_password: *const c_char,
     user_data: *mut c_void,
     network_cb_user_data: *mut c_void,
     o_network_obs_cb: unsafe extern "C" fn(*mut c_void, i32, i32),
```
Comment (nbaksalyar) [inline safe_authenticator/src/ffi/mod.rs]: Would be good to add param names here as well (for `o_network_obs_cb`)

File: safe_authenticator/src/ffi/apps.rs (hunk 5 above)
Comment (m-cat) [reply safe_authenticator/src/ffi/apps.rs]: I believe it was originally missing and added in one of my other PRs. I think this is a merge error.

File: safe_authenticator/src/ffi/mod.rs (hunk 7 above)
Comment (m-cat) [reply safe_authenticator/src/ffi/mod.rs]: Good catch, I tried not to miss anything but I guess this slipped through.

**Pull request merged**